common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -170,6 +170,13 @@ bool socket_thread::handle_cmdu_message_uds(Socket *sd)
     if (available_bytes != message_size) {
         THREAD_LOG(ERROR) << "available bytes = " << available_bytes
                           << " message_size = " << message_size;
+        // discard remaining bytes
+        ssize_t bytes_to_discard = message_size - available_bytes;
+        ssize_t bytes_discarded  = sd->discardBytes(bytes_to_discard);
```

Comment:
I don't understand why would we want to do that?
I would expect that if the message is too big, then we read what we can, and let the message handler that receives the data to know that there are more data bytes that will arrive. The handler function should handle it.
Besides that, it is probably will be a better idea to fail the building of a message if it is too big and print an error message, instead of throwing away data on the receiver side.

Anyway, If data is being thrown away, you should add an ERROR print about it.